packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -498,6 +505,7 @@ export const Table = <
     },
     useFlexLayout,
     useResizeColumns(ownerDocument),
+    useGlobalFilter,
     useFilters,
     useSubRowFiltering(hasAnySubRows),
```

Comment:
globalFilter with subrows would be fixed if you moved `useGlobalFilter` after `useSubRowFiltering`